xbmc/cores/AudioEngine/Utils/PackerMAT.h

```diff
@@ -0,0 +1,106 @@
+/*
+ *  Copyright (C) 2024 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include <deque>
+#include <stdint.h>
+#include <vector>
+
+struct TrueHDMajorSyncInfo
+{
+  int ratebits{0};
+  uint16_t outputTiming{0};
+  bool outputTimingPresent{false};
+  bool valid{false};
+};
+
+enum class Type : bool
+{
+  PADDING = true,
+  DATA = false,
+};
+
+class CPackerMAT
+{
+public:
+  CPackerMAT();
+  ~CPackerMAT() = default;
+
+  bool PackTrueHD(const uint8_t* data, int size);
+  bool HaveOutput() const { return !m_outputQueue.empty(); }
+  std::vector<uint8_t> GetOutputFrame();
+
+private:
+  struct MATState
+  {
+    bool init;
+    int ratebits;
+    uint16_t outputTiming;
+    bool outputTimingValid;
+
+    uint16_t prevFrametime;
+    bool prevFrametimeValid;
+
+    uint32_t matFramesize;
+    uint32_t prevMatFramesize;
+
+    uint32_t padding;
+    uint32_t nSamples;
+    int nSamplesOffset;
+  };
+
+  void WriteHeader();
+  void WritePadding();
+  void AppendData(const uint8_t* data, int size, Type type);
+  uint32_t GetCount() const { return m_bufferCount; }
+  int FillDataBuffer(const uint8_t* data, int size, Type type);
+  void FlushPacket();
+  static TrueHDMajorSyncInfo ParseTrueHDMajorSyncHeaders(const uint8_t* p, int buffsize);
```

Comment:
static?

Also maybe use std::optional instead of using the valid strict member?

Reply:
Not uses any member variable but can be also const method. Changed.

Reply:
> Also maybe use std::optional instead of using the valid strict member?

If `std::optional` suppose any minimal overhead I want avoid it for performance reasons.